File: Leetcode_POTD_2025/8_August/679_24_Game.cpp
class Solution {
public:
    bool judgePoint24(vector<int>& cards) {
        vector<double> cosmic_arr(cards.begin(), cards.end());
        return cosmic_quest(cosmic_arr);
    }

private:
    bool cosmic_quest(vector<double> realm) {
        if (realm.size() == 1) {
            return abs(realm[0] - 24.0) < 1e-6;
        }

        for (int i = 0; i < realm.size(); ++i) {
            for (int j = i + 1; j < realm.size(); ++j) {
                vector<double> new_realms;
                for (int k = 0; k < realm.size(); ++k) {
                    if (k != i && k != j) {
                        new_realms.push_back(realm[k]);
                    }
                }

                vector<double> temp_ops;
                temp_ops.push_back(realm[i] + realm[j]);
                temp_ops.push_back(realm[i] * realm[j]);
                temp_ops.push_back(realm[i] - realm[j]);
                temp_ops.push_back(realm[j] - realm[i]);
                if (realm[j] != 0) temp_ops.push_back(realm[i] / realm[j]);
                if (realm[i] != 0) temp_ops.push_back(realm[j] / realm[i]);

                for (double val : temp_ops) {
                    vector<double> next_realm = new_realms;
                    next_realm.push_back(val);
                    if (cosmic_quest(next_realm)) return true;
                }
            }
        }
        return false;
    }
};

File: Leetcode_POTD_2025/8_August/679_24_Game.py
import math

class Solution(object):
    def judgePoint24(self, cards):
        """
        :type cards: List[int]
        :rtype: bool
        """
        cosmic_arr = [float(card) for card in cards]
        return self._cosmic_quest(cosmic_arr)

    def _cosmic_quest(self, realm):
        if len(realm) == 1:
            return abs(realm[0] - 24.0) < 1e-6

        for i in range(len(realm)):
            for j in range(i + 1, len(realm)):
                new_realms = [realm[k] for k in range(len(realm)) if k != i and k != j]

                temp_ops = [
                    realm[i] + realm[j],
                    realm[i] * realm[j],
                    realm[i] - realm[j],
                    realm[j] - realm[i],
                ]

                if realm[j] != 0:
                    temp_ops.append(realm[i] / realm[j])
                if realm[i] != 0:
                    temp_ops.append(realm[j] / realm[i])

                for val in temp_ops:
                    next_realm = new_realms + [val]
                    if self._cosmic_quest(next_realm):
                        return True
        return False
